### PR TITLE
Updated Service Locator logging

### PR DESCRIPTION
**Pull request by StephenHodgson**
Overview
---
- Reduced service searching logs.
- Misc formatting.
- Added better handling for searching for services when shutting down.

Changes
---
- Fixes: #3077

### CODE REVIEW

File: Assets/MixedRealityToolkit/_Core/Services/MixedRealityToolkit.cs [hunk 1]
```diff
@@ -593,7 +611,7 @@ public IMixedRealityService GetService(Type type)
                 GetService(type, out service);
             }
 
-            if (service == null)
+            if (service == null && showLogs)
             {
                 Debug.Log($"Unable to find {type.Name}.");
```
Comment (StephenHodgson) [inline Assets/MixedRealityToolkit/_Core/Services/MixedRealityToolkit.cs]:
```suggestion
                Debug.LogWarning($"Unable to find {type.Name}.");
```

File: Assets/MixedRealityToolkit/_Core/Services/MixedRealityToolkit.cs [hunk 2]
```diff
@@ -550,21 +560,28 @@ public bool RegisterService(Type type, IMixedRealityService service)
...
         /// <returns>The Mixed Reality Toolkit of the specified type</returns>
-        public IMixedRealityService GetService(Type type)
+        public IMixedRealityService GetService(Type type, bool showLogs = true)
         {
+            if (isApplicationQuitting)
+            {
+                return null;
+            }
+
             if (ActiveProfile == null)
```
Comment (davidkline-ms) [inline Assets/MixedRealityToolkit/_Core/Services/MixedRealityToolkit.cs]: need to check showLogs here and other similar calls

Comment (StephenHodgson) [reply Assets/MixedRealityToolkit/_Core/Services/MixedRealityToolkit.cs]: Why? These are much more important than the other where we check. This sort of thing is pretty serious and we should def always show this error no matter what.

Comment (davidkline-ms) [reply Assets/MixedRealityToolkit/_Core/Services/MixedRealityToolkit.cs]: Is that not the purpose of the showLogs argument? If a customer passes in "showLogs == false" it seems as if the method should not log.

This is where something like the logging level task/proposal can address the "much more important". It gives customers a finer grain control over the types of messages received.

**Comment by davidkline-ms** [comment]
> Proposed changes are outside the scope of the PR and have much bigger implications.

How so? A customer asking for no logging should receive no logging.

**Comment by davidkline-ms** [comment]
The contract specified in this PR appears to be that setting showLogs to false will stop log messages. As implemented this is not the case, hence the confusion. How is this outside of the scope?

Let's take this offline and discuss in a smaller group. Historically. giving customers control over which specific log messages to see has been a net positive. It would be helpful to be able to pro/con an "all off" vs "log level" solution.

**Comment by StephenHodgson** [comment]
> How is this outside of the scope?

Because this PR is addressing the specific issue of log "spamming" as described in #3077. Anything outside of that is a further discussion, proposal, and task.

**Comment by StephenHodgson** [comment]
I'm tired of dragging out Pull Requests like this out over something we should be discussing and formally proposing and creating tasks for. We can circle back to the concerns you have after we've agreed on the best path forward. Let's not bog down the development process with petty politics.

**Comment by StephenHodgson** [comment]
Let's get this pull request completed and address the logging levels in a separate subsequent pull request.

**Pull request merged**